base/libgit2/types.jl

```diff
@@ -849,6 +853,15 @@ end
 
 with(f::Function, ::Type{T}, args...) where {T} = with(f, T(args...))
 
+"""
+    with_warn(f::Function, ::Type{T}, args..)
+
+Resource management helper function. Apply `f` to `args`, first constructing
+an instance of type `T` out of them. Makes sure to call `close` on the resulting
```

Comment:
"Out of them" sounds confusing on first read. Maybe "first constructing an instance of type `T` using the given arguments"?